### ISSUE TITLE
Type check error on result of new types

### ISSUE THREAD

**Issue by blois** [microsoft/pyright#3112]
**Describe the bug**
Error reported when accessing members of a dynamically created type.

**To Reproduce**
```python
def add(self, a, b):
  return a + b

C = type('C', (), {'add': add})
c = C()
c.add(1, 2)
```

Result:
```
sample.py
  sample.py:6:3 - error: Cannot access member "add" for type "C"
    Member "add" is unknown (reportGeneralTypeIssues)
```

**Expected behavior**
Hovering over `c.add` in VSCode with Pylance shows it has type `any`.

**VS Code extension or command-line**
Command-line with version `1.1.224`

**Comment by erictraut** [comment]
Dynamically-created classes will not work well with static type checking. Is there a reason why you're dynamically creating classes rather than using the standard class declaration syntax?

As a potential workaround, you could use an `Any` type annotation:
```python
c: Any = C()
```

**Comment by blois** [comment]
Should this be a separate bug that it's not reported as an error while editing the file?

**Comment by erictraut** [comment]
I don't think I understand. Which error is not reported?

It looks to me like it's working as expected.
